test/Altinn.App.Api.Tests/Controllers/DataController_PatchFormDataImplementation.cs

```diff
@@ -62,7 +62,7 @@ public DataController_PatchFormDataImplementation()
         _serviceCollection.AddSingleton(_dataElementValidator);
         _serviceProvider = _serviceCollection.BuildServiceProvider();
         var validationService = new ValidationService(
-            _serviceProvider,
+            new ValidatorFactory(_serviceProvider.GetServices<ITaskValidator>(), _serviceProvider.GetServices<IDataElementValidator>(), _serviceProvider.GetServices<IFormDataValidator>()),
```

Comment:
Any reason for registering services and building a serviceProvider? Mocks can, as far as I can see, be injected directly into the ValidationFactory. I think this in turn could remove the need for implementing IAsyncDisposable.

Reply:
The reason is that the previous implementation used a ServiceProivder and I did not see any reason to rewrite this testing logic.

Reply:
I think we should strive to keep the test as simple as possible for the next one visiting this part of the code

Reply:
Done